### PR TITLE
chore: set current state in ValueEntity testkit

### PR DESCRIPTION
**Pull request by octonato**
Discovered that while working on #1135

We have introduced the `currentState` for the value entities, but if a user uses it and then write a test using the testkit, they will get an error because the testkit was not setting the state.

For EventSourced entities, it's already covered in #1135. This is done in the `EventSourcedEntityEffectsRunner`

### CODE REVIEW

File: codegen/java-gen/src/main/scala/kalix/codegen/java/ValueEntityTestKitGenerator.scala [hunk 1]
```diff
@@ -144,6 +144,7 @@ object ValueEntityTestKitGenerator {
         s"""|public ValueEntityResult<$output> ${lowerFirst(command.name)}(${command.inputType.fullName} ${lowerFirst(
           command.inputType.name)}, Metadata metadata) {
        |  entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/main/scala/kalix/codegen/java/ValueEntityTestKitGenerator.scala:146]:
```suggestion
       |  entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/absolute-packages/generated-test-managed/org/example/domain/CounterTestKit.java [hunk 2]
```diff
@@ -77,12 +77,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
 
   public ValueEntityResult<Empty> increase(CounterApi.IncreaseValue increaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/absolute-packages/generated-test-managed/org/example/domain/CounterTestKit.java:79]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/absolute-packages/generated-test-managed/org/example/domain/CounterTestKit.java [hunk 3]
```diff
@@ -77,12 +77,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
...
+    entity._internalSetCurrentState(state);
     ValueEntity.Effect<Empty> effect = entity.increase(state, increaseValue);
     return interpretEffects(effect);
   }
 
   public ValueEntityResult<Empty> decrease(CounterApi.DecreaseValue decreaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/absolute-packages/generated-test-managed/org/example/domain/CounterTestKit.java:86]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/domain-in-service-package/generated-test-managed/org/example/valueentity/CounterTestKit.java [hunk 4]
```diff
@@ -75,12 +75,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
 
   public ValueEntityResult<Empty> increase(CounterApi.IncreaseValue increaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/domain-in-service-package/generated-test-managed/org/example/valueentity/CounterTestKit.java:77]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/domain-in-service-package/generated-test-managed/org/example/valueentity/CounterTestKit.java [hunk 5]
```diff
@@ -75,12 +75,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
...
+    entity._internalSetCurrentState(state);
     ValueEntity.Effect<Empty> effect = entity.increase(state, increaseValue);
     return interpretEffects(effect);
   }
 
   public ValueEntityResult<Empty> decrease(CounterApi.DecreaseValue decreaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/domain-in-service-package/generated-test-managed/org/example/valueentity/CounterTestKit.java:84]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/named-new-style/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java [hunk 6]
```diff
@@ -76,12 +76,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
 
   public ValueEntityResult<Empty> increase(CounterApi.IncreaseValue increaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/named-new-style/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java:78]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/named-new-style/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java [hunk 7]
```diff
@@ -76,12 +76,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
...
+    entity._internalSetCurrentState(state);
     ValueEntity.Effect<Empty> effect = entity.increase(state, increaseValue);
     return interpretEffects(effect);
   }
 
   public ValueEntityResult<Empty> decrease(CounterApi.DecreaseValue decreaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/named-new-style/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java:85]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/state-events-in-different-pacakge/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java [hunk 8]
```diff
@@ -77,12 +77,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
 
   public ValueEntityResult<Empty> increase(CounterApi.IncreaseValue increaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/state-events-in-different-pacakge/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java:79]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/state-events-in-different-pacakge/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java [hunk 9]
```diff
@@ -77,12 +77,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
...
+    entity._internalSetCurrentState(state);
     ValueEntity.Effect<Empty> effect = entity.increase(state, increaseValue);
     return interpretEffects(effect);
   }
 
   public ValueEntityResult<Empty> decrease(CounterApi.DecreaseValue decreaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/state-events-in-different-pacakge/generated-test-managed/org/example/valueentity/domain/CounterTestKit.java:86]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/unnamed-new-style/generated-test-managed/org/example/valueentity/CounterServiceEntityTestKit.java [hunk 10]
```diff
@@ -76,12 +76,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
 
   public ValueEntityResult<Empty> increase(CounterApi.IncreaseValue increaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/unnamed-new-style/generated-test-managed/org/example/valueentity/CounterServiceEntityTestKit.java:78]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

File: codegen/java-gen/src/test/resources/tests/value-entity/unnamed-new-style/generated-test-managed/org/example/valueentity/CounterServiceEntityTestKit.java [hunk 11]
```diff
@@ -76,12 +76,14 @@ private <Reply> ValueEntityResult<Reply> interpretEffects(ValueEntity.Effect<Rep
...
+    entity._internalSetCurrentState(state);
     ValueEntity.Effect<Empty> effect = entity.increase(state, increaseValue);
     return interpretEffects(effect);
   }
 
   public ValueEntityResult<Empty> decrease(CounterApi.DecreaseValue decreaseValue, Metadata metadata) {
     entity ._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```
Comment (efgpinto) [inline codegen/java-gen/src/test/resources/tests/value-entity/unnamed-new-style/generated-test-managed/org/example/valueentity/CounterServiceEntityTestKit.java:85]:
```suggestion
    entity._internalSetCommandContext(Optional.of(new TestKitValueEntityCommandContext(entityId, metadata)));
```

**Review by efgpinto — APPROVED** [review]
LGTM... I left some suggested changes of something that was not actually introduced by your PR, but was messing up with my OCD. Since it's in the same place, I would expect it to work right away by accepting the suggestions.

But feel free to ignore it and just merge as is if you don't wanna risk it! :)

**Comment by octonato** [comment]
@efgpinto, I applied your suggestion. Let's see what CI will tell us. Should work. 

My OCD wasn't triggered by that, because I didn't see it. 😱

**Pull request merged**